File: src/algorithmic_toolbox/week3/MoneyChange.java
package algorithmic_toolbox.week3;

import java.util.Scanner;

public class MoneyChange {
  public static final int[] DENOMINATIONS = {10, 5, 1};
  
  public static int countCoins(int money, int[] denominations) {
    int totalCoins = 0;
    for (int coinValue : denominations) {
      if (money == 0) { break; }
      
      totalCoins += money / coinValue;
      money = money % coinValue;
    }
    return totalCoins;
  }

  public static void main(String[] args) {
    Scanner scanner = new Scanner(System.in);
    int money = scanner.nextInt();
    scanner.close();

    System.out.println(countCoins(money, DENOMINATIONS));
  }
}

File: src/algorithmic_toolbox/week3/money_change.py
from typing import List

DENOMINATIONS = [10, 5, 1]


def count_coins(money: int, denominations: List[int]) -> int:
    total_coins = 0
    for coin_value in denominations:
        if money == 0:
            break

        total_coins += money // coin_value
        money = money % coin_value
    
    return total_coins


if __name__ == '__main__':
    money = int(input())
    print(count_coins(money, DENOMINATIONS))
